code/modules/antagonists/heretic/knowledge/sacrifice_knowledge/sacrifice_knowledge.dm

```diff
@@ -0,0 +1,613 @@
+
+	sac_target.visible_message("<span class='danger'>[sac_target] begins to shudder violenty as dark tendrils begin to drag them into thin air!</span>")
+	sac_target.handcuffed = new /obj/item/restraints/handcuffs/cult(sac_target)
+	sac_target.update_handcuffed()
+	if(sac_target.legcuffed)
+		sac_target.clear_legcuffs(TRUE)
+
+	sac_target.setBrainLoss(40)
+	sac_target.do_jitter_animation()
+
+	addtimer(CALLBACK(sac_target, TYPE_PROC_REF(/mob/living/carbon, do_jitter_animation)), SACRIFICE_SLEEP_DURATION * (1/3))
+	addtimer(CALLBACK(sac_target, TYPE_PROC_REF(/mob/living/carbon, do_jitter_animation)), SACRIFICE_SLEEP_DURATION * (2/3))
+
+	// If our target is dead, try to revive them
+	// and if we fail to revive them, don't proceede the chain
+	sac_target.adjustOxyLoss(-100, FALSE)
+	if(!sac_target.heal_and_revive(50, "<span class='danger'>[sac_target]'s heart begins to beat with an unholy force as they return from death!</span>"))
+		return
+
+	if(sac_target.AdjustSleeping(SACRIFICE_SLEEP_DURATION))
+		to_chat(sac_target, "<span class='hierophant_warning'>Your mind feels torn apart as you fall into a shallow slumber...</span>")
+	else
+		to_chat(sac_target, "<span class='hierophant_warning'>Your mind begins to tear apart as you watch dark tendrils envelop you.</span>")
+
+	sac_target.AdjustWeakened(SACRIFICE_SLEEP_DURATION * 1.2)
+	sac_target.AdjustImmobilized(SACRIFICE_SLEEP_DURATION * 1.2)
+
+	addtimer(CALLBACK(src, PROC_REF(after_target_sleeps), sac_target, destination), SACRIFICE_SLEEP_DURATION * 0.5) // Teleport to the minigame
+
+	return TRUE
+
+/**
+ * This proc is called from [proc/begin_sacrifice] after the [sac_target] falls asleep), shortly after the sacrifice occurs.
+ *
+ * Teleports the [sac_target] to the heretic room, asleep.
+ * If it fails to teleport, they will be disemboweled and stop the chain.
+ *
+ * Arguments
+ * * sac_target - the mob being sacrificed.
+ * * destination - the spot they're being teleported to.
+ */
+/datum/heretic_knowledge/hunt_and_sacrifice/proc/after_target_sleeps(mob/living/carbon/human/sac_target, turf/destination)
+	if(QDELETED(sac_target))
+		return
+
+	// The target disconnected or something, we shouldn't bother sending them along.
+	if(!sac_target.client || !sac_target.mind)
+		disembowel_target(sac_target)
+		return
+
+	curse_organs(sac_target)
+
+	// Send 'em to the destination. If the teleport fails, just disembowel them and stop the chain
+	if(!destination || SEND_SIGNAL(sac_target, COMSIG_MOVABLE_TELEPORTING, destination) & COMPONENT_BLOCK_TELEPORT)
+		disembowel_target(sac_target)
+		return
+	sac_target.forceMove(destination)
+	// If our target died during the (short) wait timer,
+	// and we fail to revive them (using a lower number than before),
+	// just disembowel them and stop the chain
+	sac_target.adjustOxyLoss(-100, FALSE) //Qwertodo: (after tm progress), if people just stab people trying to kill them as they revive, give them damage resistance
```

Comment:
more todo

Reply:
in my defense this one says after tm progress